src/everest/everest_storage.py

```diff
@@ -0,0 +1,880 @@
+                    config.variables.upper_bounds, dtype=polars.Float64
+                ),
+            }
+        )
+
+        self.data.objective_functions = polars.DataFrame(
+            {
+                "objective_name": config.objectives.names,
+                "weight": polars.Series(
+                    config.objectives.weights, dtype=polars.Float64
+                ),
+                "normalization": polars.Series(  # Q: Is this correct?
+                    [1.0 / s for s in config.objectives.scales],
+                    dtype=polars.Float64,
+                ),
+            }
+        )
+
+        if config.nonlinear_constraints is not None:
+            self.data.nonlinear_constraints = polars.DataFrame(
+                {
+                    "constraint_name": config.nonlinear_constraints.names,
+                    "normalization": [
+                        1.0 / s for s in config.nonlinear_constraints.scales
+                    ],  # Q: Is this correct?
+                    "constraint_rhs_value": config.nonlinear_constraints.rhs_values,
+                    "constraint_type": config.nonlinear_constraints.types,
+                }
+            )
+
+        self.data.realization_weights = polars.DataFrame(
+            {
+                "realization": polars.Series(
+                    config.realizations.names, dtype=polars.UInt32
+                ),
+                "weight": polars.Series(
+                    config.realizations.weights, dtype=polars.Float64
+                ),
+            }
+        )
+
+    def _store_function_results(self, results: FunctionResults) -> _EvaluationResults:
+        # We could select only objective values,
+        # but we select all to also get the constraint values (if they exist)
+        realization_objectives = polars.from_pandas(
+            results.to_dataframe(
+                "evaluations",
+                select=["objectives", "evaluation_ids"],
+            ).reset_index(),
+        ).select(
+            "batch_id",
+            "realization",
+            "objective",
+            "objectives",
+            "evaluation_ids",
+        )
+
+        if results.nonlinear_constraints is not None:
+            realization_constraints = polars.from_pandas(
+                results.to_dataframe(
+                    "evaluations",
+                    select=["constraints", "evaluation_ids"],
+                ).reset_index(),
+            ).select(
+                "batch_id",
+                "realization",
+                "evaluation_ids",
+                "nonlinear_constraint",
+                "constraints",
+            )
+
+            realization_constraints = self._rename_ropt_df_columns(
+                realization_constraints
+            )
+
+            batch_constraints = polars.from_pandas(
+                results.to_dataframe("nonlinear_constraints").reset_index()
+            ).select("batch_id", "nonlinear_constraint", "values")
```

Comment:
Don't use the `nonlinear_constraints` field. It is used to store the difference between the constraint and its rhs value, not the total constraint itself. Also, this field is likely to disappear, since current contraint handling is to be revised in `ropt`.

Instead, get the values from the `functions` fields, see comment below.